scipy/interpolate/interpolate.py

```diff
@@ -497,10 +499,6 @@ def fill_value(self):
     def fill_value(self, fill_value):
         # extrapolation only works for nearest neighbor and linear methods
         if _do_extrapolate(fill_value):
-            if self._kind not in ('nearest', 'linear'):
-                raise ValueError("Extrapolation does not work with "
-                                 "kind=%s" % self._kind)
-
```

Comment:
Both 0.17.1 and this accept bad input in constructor and raise on first use:

```
In [12]: interp1d(np.arange(8), np.arange(8),
   ....:          fill_value="bogus", bounds_error=False)
Out[12]: <scipy.interpolate.interpolate.interp1d at 0x7fb00fd72c00>

In [13]: interp1d(np.arange(8), np.arange(8),
         fill_value="bogus", bounds_error=False)(11)
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
<ipython-input-13-400eb931e1f1> in <module>()
      1 interp1d(np.arange(8), np.arange(8),
----> 2          fill_value="bogus", bounds_error=False)(11)

/home/br/repos/scipy/build/testenv/lib/python2.7/site-packages/scipy/interpolate/polyint.pyc in __call__(self, x)
     77         """
     78         x, x_shape = self._prepare_x(x)
---> 79         y = self._evaluate(x)
     80         return self._finish_y(y, x_shape)
     81 

/home/br/repos/scipy/build/testenv/lib/python2.7/site-packages/scipy/interpolate/interpolate.pyc in _evaluate(self, x_new)
    595                 # and flattened to work here
    596                 y_new[below_bounds] = self._fill_value_below
--> 597                 y_new[above_bounds] = self._fill_value_above
    598         return y_new
    599 

ValueError: could not convert string to float: bogus
```

Do you want me to change it to raise in the constructor?